### PR TITLE
fix test_node_request_snapshot_reject_merge

### PR DESCRIPTION
**Pull request by 5kbpers**
Signed-off-by: 5kbpers <tangminghua@pingcap.com>

## What have you changed? (mandatory)

test cases::test_merge::test_node_request_snapshot_reject_merge failed while running CI (see https://github.com/tikv/tikv/issues/5054).
I found it may be caused by the variable `notifier` https://github.com/tikv/tikv/blob/c3a8f2152164abe7c2285ee836d087146c7c9422/tests/failpoints/cases/test_merge.rs#L444-L451  sends a message after the variable `rx` was deconstructed. https://github.com/tikv/tikv/blob/c3a8f2152164abe7c2285ee836d087146c7c9422/tests/failpoints/cases/test_merge.rs#L454
Deconstructing `cluster` before the function returns could fix this problem.

## What are the type of the changes? (mandatory)

- Bug fix (change which fixes an issue)

### CODE REVIEW

**Comment by siddontang** [comment]
> it may be caused 

can we ensure it?

**Comment by 5kbpers** [comment]
> > it may be caused
> 
> can we ensure it?

@siddontang I think the backtraces show it is the reason for test failing.

File: tests/failpoints/cases/test_merge.rs [hunk 1]
```diff
@@ -467,4 +467,5 @@ fn test_node_request_snapshot_reject_merge() {
     let target = cluster.get_region(target_region.get_start_key());
     assert_eq!(target, target_region);
     fail::remove(region_gen_snap_fp);
+    drop(cluster);
```
Comment (overvenus) [inline tests/failpoints/cases/test_merge.rs]: But cluster is dropped automatically at the end of the test.

Comment (5kbpers) [reply tests/failpoints/cases/test_merge.rs]: Yes, but seems it sends a message to `rx` after `rx` is dropped, then causes a `SendError`. So I think ensuring `cluster` is dropped earlier than `rx` would solve this problem.

Comment (overvenus) [reply tests/failpoints/cases/test_merge.rs]: LGTM, could you add some comments?

Comment (5kbpers) [reply tests/failpoints/cases/test_merge.rs]: Sure.

**Comment by siddontang** [comment]
PTAL @Connor1996

**Pull request merged**